frontend/simple/views/components/Voting/Voting.vue

```diff
@@ -130,36 +153,130 @@ export default {
   },
   data () {
     return {
-      closeProposalState: {
+      ephemeral: {
         state: null,
         countdown: null
       }
     }
   },
   computed: {
+    ctas () {
+      const { value, originalValue } = this
+      const { TypeInvitation, TypeRemoval } = HashableGroupProposal
+      if (this.type === TypeInvitation) {
+        return {
+          for: template(
+            L('Invite {value}'), { value }
+          ),
+          against: template(
+            L('Don\'t invite'), { value }
+          )
+        }
+      } else if (this.type === TypeRemoval) {
+        return {
+          for: template(
+            L('Remove {value}'), { value }
+          ),
+          against: template(
+            L('Keep {value}'), { value }
+          )
+        }
+      } else {
+        return {
+          for: template(
+            L('Change to {value}'), { value }
+          ),
+          against: template(
+            L('Keep {originalValue}'), { originalValue }
+          )
+        }
+      }
```

Comment:
This block could be improved by making it much shorter (29 vs 17 lines; also fixed a bug where `template` isn't used):

```js
switch (this.type) {
  case HashableGroupProposal.TypeInvitation:
    return {
      for: template(L('Invite {value}'), { value }),
      against: L('Don\'t invite'),
    }
  case HashableGroupProposal.TypeRemoval:
    return {
      for: template(L('Remove {value}'), { value }),
      against: template(L('Keep {value}'), { value })
    }
  default:
    return {
      for: template(L('Change to {value}'), { value }),
      against: template(L('Keep {originalValue}'), { originalValue })
    }
}
```

Reply:
And per the comments below, by pulling out the `const` types to the top of the file, you can change my `HashableGroupProposal.TypeInvitation` above to `TypeInvitation` without having to re-pull them out.